clients/src/main/java/org/apache/kafka/common/requests/OffsetFetchRequest.java

```diff
@@ -227,6 +227,24 @@ public boolean requireStable() {
         return data.requireStable();
     }
 
+    public List<OffsetFetchRequestData.OffsetFetchRequestGroup> groups() {
+        if (version() >= 8) {
+            return data.groups();
+        } else {
+            OffsetFetchRequestData.OffsetFetchRequestGroup group =
+                new OffsetFetchRequestData.OffsetFetchRequestGroup().setGroupId(data.groupId());
+
+            data.topics().forEach(topic -> {
```

Comment:
Do you think this can be a bit more intuitive?
```
group.topics.addAll(
  data.topics().stream().map(new Offset...).collect(Collections.toList())
)
```

Reply:
Is this doing the same thing?

Reply:
I agree with Justine. The intermediate collection is not needed.